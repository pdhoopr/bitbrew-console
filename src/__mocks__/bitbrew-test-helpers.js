import { shallow } from 'enzyme';
import enzymeSerializer from 'enzyme-to-json/serializer';
import diff from 'jest-diff';
import { addSerializer, getSerializers, utils } from 'jest-snapshot';
import styledSerializer from 'jest-styled-components/src/styleSheetSerializer';
import stripAnsi from 'strip-ansi';

const DIFF_HEADER = 'Diff:\n';

export const diffComponents = (baseComponent, compareComponent) => {
  const serializers = getSerializers();
  [styledSerializer, enzymeSerializer].forEach(serializer => {
    if (!serializers.includes(serializer)) {
      addSerializer(serializer);
    }
  });
  const base = `${utils.serialize(shallow(baseComponent))}\n`;
  const compare = `${utils.serialize(shallow(compareComponent))}\n`;
  const changes = diff(base, compare, {
    aAnnotation: `<BaseComponent />`,
    bAnnotation: `<CompareComponent />`,
    contextLines: 3,
    expand: false,
  });
  return `${DIFF_HEADER}${changes}`;
};

export const diffSerializer = {
  test(value) {
    return typeof value === 'string' && value.startsWith(DIFF_HEADER);
  },
  print(value) {
    return stripAnsi(value).trim();
  },
};

export const fixtures = {
  accessToken:
    'eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJPbmxpbmUgSldUIEJ1aWxkZXIiLCJpYXQiOjE1MjUxMTUzNTAsImV4cCI6MTU1NjY1MTM1MCwiYXVkIjoid3d3LmV4YW1wbGUuY29tIiwic3ViIjoianJvY2tldEBleGFtcGxlLmNvbSJ9.4O-W8aeOOUFU8fkGTAFaAu9D9nK_l61yPomNG1Qkny8',
  orgs: {
    danlaw: {
      id: 'bd006616-3bd2-4783-a84d-b0fa4ed81135',
      properName: 'Danlaw',
    },
    azuga: {
      id: '8d9ec637-f3bb-4c46-83ee-11764d09f0b2',
      properName: 'Azuga',
    },
    bitbrew: {
      id: '94d9b0ca-a758-4502-8dcd-41d883e38714',
      properName: 'BitBrew',
    },
  },
  users: {
    bill: {
      email: 'bill@bitbrew.com',
      name: 'Bill Monaghan',
    },
    cynthia: {
      email: 'cynthia@bitbrew.com',
      name: 'Cynthia Hamilton',
    },
    alex: {
      email: 'bill@bitbrew.com',
      name: 'Alex Li',
    },
  },
};
