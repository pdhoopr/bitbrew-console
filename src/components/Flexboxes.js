import styled, { css } from 'styled-components';

export const baseStyles = css`
  align-items: center;
  display: flex;
`;

export const FlexBetween = styled.div`
  ${baseStyles};
  justify-content: space-between;
`;

export const FlexCenter = styled.div`
  ${baseStyles};
  justify-content: center;
`;

export const FlexEnd = styled.div`
  ${baseStyles};
  justify-content: flex-end;
`;

export const FlexStart = styled.div`
  ${baseStyles};
  justify-content: flex-start;
`;
