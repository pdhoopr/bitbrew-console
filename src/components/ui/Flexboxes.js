import styled, { css } from "styled-components";

const baseStyles = css`
  align-items: center;
  display: flex;

  > :not(:last-child) {
    margin-right: ${({ gap }) => gap && "var(--size-16)"};
  }
`;

export const FlexBetween = styled.div`
  ${baseStyles};
  justify-content: space-between;
`;

export const FlexCenter = styled.div`
  ${baseStyles};
  justify-content: center;
`;

export const FlexEnd = styled.div`
  ${baseStyles};
  justify-content: flex-end;
`;

export const FlexStart = styled.div`
  ${baseStyles};
  justify-content: flex-start;
`;
