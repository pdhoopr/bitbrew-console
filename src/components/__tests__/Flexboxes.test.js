import { diffComponents } from 'bitbrew-test-helpers';
import { shallow } from 'enzyme';
import React from 'react';
import styled from 'styled-components';
import { FlexBetween, FlexCenter, FlexEnd, baseStyles } from '../Flexboxes';

const BaseFlex = styled.div`
  ${baseStyles};
`;

test('renders DOM shared by all flex containers', () => {
  const dom = shallow(<BaseFlex />);
  expect(dom).toMatchSnapshot();
});

test('renders DOM specific to a flex container with space between elements', () => {
  const base = <BaseFlex />;
  const comp = <FlexBetween />;
  const diff = diffComponents(base, comp);
  expect(diff).toMatchSnapshot();
});

test('renders DOM specific to a flex container with elements aligned in the center', () => {
  const base = <BaseFlex />;
  const comp = <FlexCenter />;
  const diff = diffComponents(base, comp);
  expect(diff).toMatchSnapshot();
});

test('renders DOM specific to a flex container with elements aligned at the end', () => {
  const base = <BaseFlex />;
  const comp = <FlexEnd />;
  const diff = diffComponents(base, comp);
  expect(diff).toMatchSnapshot();
});
