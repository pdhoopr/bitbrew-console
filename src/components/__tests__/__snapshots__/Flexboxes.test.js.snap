// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders DOM shared by all flex containers 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

<div
  className="c0"
/>
`;

exports[`renders DOM specific to a flex container with elements aligned at the end 1`] = `
Diff:
- <BaseComponent />
+ <CompareComponent />

@@ -8,6 +8,10 @@
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
+   -webkit-box-pack: end;
+   -webkit-justify-content: flex-end;
+   -ms-flex-pack: end;
+   justify-content: flex-end;
  }
  
  <div
`;

exports[`renders DOM specific to a flex container with elements aligned in the center 1`] = `
Diff:
- <BaseComponent />
+ <CompareComponent />

@@ -8,6 +8,10 @@
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
+   -webkit-box-pack: center;
+   -webkit-justify-content: center;
+   -ms-flex-pack: center;
+   justify-content: center;
  }
  
  <div
`;

exports[`renders DOM specific to a flex container with space between elements 1`] = `
Diff:
- <BaseComponent />
+ <CompareComponent />

@@ -8,6 +8,10 @@
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
+   -webkit-box-pack: justify;
+   -webkit-justify-content: space-between;
+   -ms-flex-pack: justify;
+   justify-content: space-between;
  }
  
  <div
`;
